### PR TITLE
fix: avoid mobile composer overlap with tappable toasts

### PR DESCRIPTION
**Pull request by kristofferremback**
## Problem

On mobile, toast notifications were rendered near the bottom of the viewport, which can overlap the message composer and block one of the most important actions in the app. The current UX also lacked a simple tap-to-dismiss path for users who want to clear toasts quickly.

## Solution

Updated the shared frontend toaster wrapper to make toast behavior mobile-aware and dismissible by tap.

### How it works

1. The toaster now resolves a default position based on breakpoint:
   - mobile: `top-center`
   - desktop: `bottom-right`
2. Mobile toasts now use safe-area-aware offsets so they sit below notches and status bars.
3. A delegated click handler on the toaster root maps the clicked toast DOM node to the active Sonner toast id and calls `toast.dismiss(id)`.
4. Clicks on interactive controls (`button`, links, form controls, role button) are ignored so action/cancel/close controls keep their existing behavior.
5. Non-dismissible toasts are explicitly ignored by the tap handler.

### Key design decisions

**1. Move mobile toasts to `top-center` instead of tweaking bottom offsets**

A top position removes overlap risk with the message composer entirely, including cases where keyboard/safe-area changes alter bottom layout.

**2. Implement dismiss-on-tap in the shared toaster wrapper**

Using delegated handling in the wrapper avoids touching every toast call site and keeps behavior consistent across the app.

**3. Preserve toast actions and non-dismissible semantics**

The handler filters interactive targets and checks `data-dismissible` to prevent accidental dismissals and preserve current UX contracts.

## New files

| File | Purpose |
| --- | --- |
| `apps/frontend/src/components/ui/sonner.test.tsx` | Adds focused tests for mobile positioning and tap-to-dismiss behavior. |

## Modified files

| File | Change |
| --- | --- |
| `apps/frontend/src/components/ui/sonner.tsx` | Added mobile-aware default placement, safe-area mobile offsets, delegated tap-to-dismiss logic, and dismissibility cursor styling. |

## Deleted files

None.

## Test plan

- [x] `bunx vitest run src/components/ui/sonner.test.tsx` (from `apps/frontend`)
- [x] `bun run --cwd apps/frontend typecheck`
- [x] Pre-commit hooks passed (lint + typecheck suites)
- [ ] Manual on-device check: trigger success/error toast on mobile and verify it appears at top and tap on body dismisses it without blocking composer input

---

🤖 _PR by Codex_

## Summary by CodeRabbit

* **New Features**
  * Toasts now position by device (top-center on mobile, bottom-right on desktop).
  * Click-to-dismiss enabled for dismissible toasts; cursor indicates dismissible items.

* **Bug Fixes**
  * Action button clicks no longer dismiss toasts.
  * Non-dismissible toasts are preserved when clicked.

* **Tests**
  * Added comprehensive tests covering positioning, dismissal behavior, and interaction edge cases.

### CODE REVIEW

File: apps/frontend/src/components/ui/sonner.tsx [hunk 1]
```diff
@@ -1,15 +1,76 @@
...
+      if (!indexRaw || !y || !x) return
+
+      const index = Number.parseInt(indexRaw, 10)
+      if (!Number.isFinite(index)) return
+
+      const clickedPosition = `${y}-${x}` as ToastPosition
+      const toastsAtPosition = filteredToasts.filter(
+        (activeToast) => (activeToast.position ?? defaultPosition) === clickedPosition
+      )
+      const clickedToast = toastsAtPosition[index]
```
[the remark this answers is not in the document]
Comment (kristofferremback) [reply apps/frontend/src/components/ui/sonner.tsx]: Fixed. I now precompute a position-aware mapping from rendered toast elements to toast IDs using a `WeakMap`, and click dismissal uses that mapping (with a dataset fallback). This avoids cross-position index ambiguity and keeps dismissal aligned with Sonner's per-position indexing.

🤖 _Response by [Claude Code](https://claude.com/claude-code)_

File: apps/frontend/src/components/ui/sonner.test.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+  it("dismisses a toast when tapping its body", async () => {
+    const user = userEvent.setup()
+    mockState.toasts = [{ id: "toast_1" }, { id: "toast_2" }]
+
+    render(<Toaster />)
+    await user.click(screen.getByText("toast_2"))
+
+    expect(mockState.dismissToast).toHaveBeenCalledWith("toast_2")
+  })
```
[the remark this answers is not in the document]
Comment (kristofferremback) [reply apps/frontend/src/components/ui/sonner.test.tsx]: Added. `sonner.test.tsx` now includes a multi-position scenario (`top-center` + `bottom-right`) that verifies clicking `top_2` dismisses `top_2` and not `top_1`.

🤖 _Response by [Claude Code](https://claude.com/claude-code)_
